satpy/tests/reader_tests/test_avhrr_l1b_gaclac.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2009-2019 Pytroll developpers
+
+# This file is part of satpy.
+
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+from unittest import TestCase, main, TestLoader, TestSuite
+import numpy as np
+import mock
+
+GAC_PATTERN = 'NSS.GHRR.{platform_id:2s}.D{start_time:%y%j.S%H%M}.E{end_time:%H%M}.B{orbit_number:05d}{end_orbit_last_digits:02d}.{station:2s}'  # noqa
+
+EXAMPLE_FILENAMES = ['NSS.GHRR.NA.D79184.S1150.E1337.B0008384.WI',
+                     'NSS.GHRR.NA.D79184.S2350.E0137.B0008384.WI',
+                     'NSS.GHRR.NA.D80021.S0927.E1121.B0295354.WI',
+                     'NSS.GHRR.NA.D80021.S1120.E1301.B0295455.WI',
+                     'NSS.GHRR.NA.D80021.S1256.E1450.B0295556.GC',
+                     'NSS.GHRR.NE.D83208.S1219.E1404.B0171819.WI',
+                     'NSS.GHRR.NG.D88002.S0614.E0807.B0670506.WI',
+                     'NSS.GHRR.TN.D79183.S1258.E1444.B0369697.GC',
+                     'NSS.GHRR.TN.D80003.S1147.E1332.B0630506.GC',
+                     'NSS.GHRR.TN.D80003.S1328.E1513.B0630507.GC',
+                     'NSS.GHRR.TN.D80003.S1509.E1654.B0630608.GC']
+
+
+class TestGACLACFile(TestCase):
+
+    def setUp(self):
+        self.pygac = mock.MagicMock()
+        self.fhs = mock.MagicMock()
+        modules = {
+            'pygac': self.pygac,
+            'pygac.gac_klm': self.pygac.gac_klm,
+            'pygac.gac_pod': self.pygac.gac_pod,
+        }
+
+        self.module_patcher = mock.patch.dict('sys.modules', modules)
+        self.module_patcher.start()
+
+    def tearDown(self):
+        self.module_patcher.stop()
+
+    def test_gaclacfile(self):
```

Comment:
Could you add docstrings on the classes and methods in here?

Reply:
Done